src/lib/shady-render.ts

```diff
@@ -125,8 +125,13 @@ const shadyRenderSet = new Set<string>();
  * output.
  */
 const prepareTemplateStyles =
-    (renderedDOM: DocumentFragment, template: Template, scopeName: string) => {
+    (scopeName: string, renderedDOM: DocumentFragment, template?: Template) => {
       shadyRenderSet.add(scopeName);
+      // If `renderedDOM` is stamped from a Template, then we need to edit that
+      // Template's underlying template element. Otherwise, we create one here
+      // to give to ShadyCSS, which still requires one while scoping.
+      const templateElement =
+          template ? template.element : document.createElement('template');
```

Comment:
Always prefer explicit comparisons over implicity ToBoolean conversions, here and in two `if` statements below.

Reply:
Ok, these are explicit now.